src/providers/Theme/Theme.provider.jsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState, useContext, useCallback } from 'react';
+
+const ThemeContext = React.createContext(null);
+
+function useTheme() {
+  const context = useContext(ThemeContext);
+  if (!context) {
+    throw new Error(`Can't use "useTheme" without a ThemeProvider.`);
+  }
+  return context;
+}
+
+function ThemeProvider({ children }) {
+  const [displayTheme, setDisplayTheme] = useState('light');
```

Comment:
useReducer is prefered for updating context state: https://stackoverflow.com/questions/54646553/usestate-vs-usereducer#:~:text=useReducer%20is%20usually%20preferable%20to,dispatch%20down%20instead%20of%20callbacks.